clients/src/main/java/org/apache/kafka/clients/consumer/internals/Fetcher.java

```diff
@@ -972,16 +975,28 @@ private Record nextFetchedRecord() {
         private List<ConsumerRecord<K, V>> fetchRecords(int maxRecords) {
             if (isFetched)
                 return Collections.emptyList();
+            if (nextInlineException != null) {
```

Comment:
With this behavior, the next call to `poll()` will advance beyond the record (or record batch) which encountered the error. I'm not sure that's what we want. Previously we expect users to manually seek past the offset with the error. This is easier said than done (https://issues.apache.org/jira/browse/KAFKA-4740), but I think we should try to preserve the current behavior here. In that case, we could simply close the record stream and set `isFetched` to true.